### DISCUSSION TITLE
Warning! Detected 437.83 seconds time difference between your browser and the server. Prometheus relies on accurate time and time drift might cause unexpected query results.

### DISCUSSION THREAD

**Discussion by hubcomsix** [prometheus/prometheus#14616] [Q&A]
### What did you do?

# PossibleSolution 
Prometheus synchronizes time with grafana to maintain the same time 

### What did you see instead? Under which circumstances?

# CurrentBehavior
In Prometheus web there is a 5min difference with Grafana

**Comment by chaofanx** [comment]
Check if there is a time gap between your server and browser

**Comment by beorn7** [comment]
This doesn't seem to be a bug report or feature request. I'll convert it into a discussion.

**Comment by bboreham** [comment]
Both programs should be taking the time from the underlying operating system, and all popular operating systems have a way to synchronize time built in. (The underlying standard is called “NTP”.)

Can you expand on why Prometheus should need more than this?

**Comment by anthonyeleven** [comment]
This didn't happen in the expression browser with older Prom releases, say ones before the Promlens integration.  I get it all the time now.

It *smells* like the server time is compared against the time the browser first loads the page, since a reload of the page makes it go away.

**Comment by bboreham** [comment]
@anthonyeleven that is an interesting thought.

To check, I opened the ["PromBench Meta" Prometheus](http://prombench.prometheus.io/prometheus-meta/query?g0.expr=go_info&g0.show_tree=0&g0.tab=table&g0.range_input=1h&g0.res_type=auto&g0.res_density=medium&g0.display_mode=lines&g0.show_exemplars=0) then went for a walk.  After 44 minutes it answers queries with no warnings.

Can you provide instructions to reproduce using that Prometheus?  I don't know what you mean by "expression browser".

**Comment by anthonyeleven** [comment]
The page I get at http://localhost:9090/query 

where I enter a PromQL expression and see the results in tabular or graphical format.

The instance at the URL provide looks diffrerent.

To reproduce simply leave the tab up for a couple of weeks without reloading, then adjust or add a panel.

**Comment by bboreham** [comment]
That’s your local machine, which gives a lot more variables to ponder. 

I think, given that approximately two people have noted this in the last two years, that it is most likely something specific about your environments.

**Reply by anthonyeleven** [reply]
Possible.  It's a company laptop with all manner of lockdown.
